instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/HttpServerRequestMetrics.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.instrumenter;
+
+import com.google.auto.value.AutoValue;
+import io.opentelemetry.api.common.AttributeType;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.metrics.DoubleValueRecorder;
+import io.opentelemetry.api.metrics.LongUpDownCounter;
+import io.opentelemetry.api.metrics.Meter;
+import io.opentelemetry.api.metrics.common.Labels;
+import io.opentelemetry.api.metrics.common.LabelsBuilder;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.ContextKey;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class HttpServerRequestMetrics implements RequestMetrics {
+
+  private static final ContextKey<State> HTTP_SERVER_REQUEST_METRICS_STATE =
+      ContextKey.named("http-server-request-metrics-state");
+
+  private static final Logger logger = LoggerFactory.getLogger(HttpServerRequestMetrics.class);
+
+  public static RequestMetricsFactory factory() {
+    return HttpServerRequestMetrics::new;
+  }
+
+  private final LongUpDownCounter activeRequests;
+  private final DoubleValueRecorder duration;
+
+  private HttpServerRequestMetrics(Meter meter) {
+    activeRequests =
+        meter
+            .longUpDownCounterBuilder("http.server.active_requests")
+            .setUnit("requests")
+            .setDescription("The number of concurrent HTTP requests that are currently in-flight")
+            .build();
+
+    duration =
+        meter
+            .doubleValueRecorderBuilder("http.server.duration")
+            .setUnit("milliseconds")
+            .setDescription("The duration of the inbound HTTP request")
+            .build();
+  }
+
+  @Override
+  public Context start(Context context, Attributes requestAttributes) {
+    long startTimeNanos = System.nanoTime();
+    Labels activeRequestLabels = activeRequestLabels(requestAttributes);
+    Labels durationLabels = durationLabels(requestAttributes);
+    activeRequests.add(1, activeRequestLabels);
+
+    return context.with(
+        HTTP_SERVER_REQUEST_METRICS_STATE,
+        new AutoValue_HttpServerRequestMetrics_State(
+            activeRequestLabels, durationLabels, startTimeNanos));
+  }
+
+  @Override
+  public void end(Context context, Attributes responseAttributes) {
+    State state = context.get(HTTP_SERVER_REQUEST_METRICS_STATE);
+    if (state == null) {
+      logger.debug(
+          "No state present when ending context {}. Cannot reset HTTP request metrics.", context);
+      return;
+    }
+    activeRequests.add(-1, state.activeRequestLabels());
+    duration.record(System.nanoTime() - state.startTimeNanos(), state.durationLabels());
+  }
+
+  private static Labels activeRequestLabels(Attributes attributes) {
+    LabelsBuilder labels = Labels.builder();
+    attributes.forEach(
```

Comment:
It'd be great if the SDK included some easy Attributes<->Labels conversion. Or if we could use Attributes directly (well, a subset of them) in metrics API...

Reply:
Yeah that's what I'm hoping for.